scr/components/App.jsx

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useState } from "react";
+
+import { Key } from "./Key";
+import { Button } from "./Button";
+
+import { Api } from "../utils/Api";
+
+export const App = () => {
+  const [keys, setKeys] = useState([]);
+  const [isLoading, setLoadingState] = useState(false);
+  const [countUsedKeys, setCountUsedKeys] = useState(0);
+
+  const toggleLoading = useCallback(() => {
+    setLoadingState(!isLoading);
+  }, [isLoading]);
+
+  const incrementUsedKeys = () => {
+    setCountUsedKeys((prevValue) => prevValue++);
+  }
+
+  const decrementUsedKeys = useCallback(() => {
+    setCountUsedKeys((prevValue) => prevValue--);
+  }, []);
+
+  const addKey = useCallback(async () => {
+    toggleLoading();
+
+    const key = await Api.generateKey();
+    setKeys((prevKeys) => prevKeys.push(key));
```

Comment:
Здесь нужно вернуть новое значение keys, а не изменить старый массив. В текущей реализации в keys запишется длина нового массива
```suggestion
    setKeys((prevKeys) => [...prevKeys, key]);
```